File: 78/p78.py
import resource, sys

resource.setrlimit(resource.RLIMIT_STACK, (resource.RLIM_INFINITY, resource.RLIM_INFINITY))

sys.setrecursionlimit(61000)

MAX = 60000

partitions = [-1]*(MAX + 1)
partitions[0] = 1

def partition(n):
	if n < 0:
		return 0
	if partitions[n] == -1:
		partitions[n] = 0
		k = 1
		while True:
			if n < k*(3*k - 1)//2:
				break
			a = partition(n - (k*(3*k - 1))//2)
			if k%2 == 0:
				partitions[n] -= a
			else:
				partitions[n] += a
			if n < k*(3*k + 1)//2:
				break
			a = partition(n - (k*(3*k + 1))//2)
			if k%2 == 0:
				partitions[n] -= a
			else:
				partitions[n] += a
			k += 1
			partitions[n] %= 1000000
	return partitions[n]

n = 1
while partition(n):
	n += 1
print(n)

File: 78/p78.c
#include <stdio.h>
#include <inttypes.h>

#define MAX 416

uint64_t partitions[MAX + 1] = {[0] = 1};

uint64_t partition(int n){
	if(n < 0){
		return 0;
	}
	if(!partitions[n]){
		for(int k = 1;; ++k){
			int a = (k&1?1:-1)*(partition(n - ((k*(3*k - 1))>>1)));
			if(!a){
				break;
			}
			partitions[n] += a;
			a = (k&1?1:-1)*(partition(n - ((k*(3*k + 1))>>1)));
			if(!a){
				break;
			}
			partitions[n] += a;
		}
	}
	return partitions[n];
}

int main(){
	printf("%"PRIu64"\n", partition(MAX));
}

